device/core/src/pnp/command_response.ts

```diff
@@ -0,0 +1,28 @@
+import { JSONValue } from '.';
+import { DeviceMethodResponse } from '../device_method';
+import { ErrorCallback } from 'azure-iot-common';
+
+export class CommandResponse {
+    private methodSend: (status: number, payload?: any | ErrorCallback, done?: ErrorCallback) => Promise<void> | void;
+
+    constructor(methodResponse: DeviceMethodResponse) {
+        this.methodSend = methodResponse.send.bind(methodResponse);
```

Comment:
>methodResponse.send.bind(methodResponse); [](http://example.com/codeflow?start=26&length=41)

Is the `bind` required here?  I assume it is, but I'm wondering why.

Reply:
When you assign a function value to a variable like this, it looses its context. Using `bind()` here will assigns `this` so that `methodResponse.send()` works correctly.